src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -155,11 +155,26 @@ public function connect()
 
             $options = $this->getOptions();
 
-            // use a memory database if the option was specified
-            if (!empty($options['memory']) || $options['name'] === static::MEMORY) {
-                $dsn = 'sqlite:' . static::MEMORY;
+            if (PHP_VERSION_ID < 80100 && (!empty($options['mode']) || !empty($options['cache']))) {
+                throw new RuntimeException('SQLite URI support requires PHP 8.1.');
+            } elseif ((!empty($options['mode']) || !empty($options['cache'])) && !empty($options['memory'])) {
+                throw new RuntimeException('Memory must not be set when cache or mode are.');
+            } elseif (PHP_VERSION_ID >= 80100) {
+                $params = [];
+                if (!empty($options['cache'])) {
+                    $params[] = 'cache=' . $options['cache'];
+                }
+                if (!empty($options['mode'])) {
+                    $params[] = 'mode=' . $options['mode'];
+                }
+                $dsn = 'sqlite:file:' . $options['name'] . '?' . implode('&', $params);
```

Comment:
Are we requiring that `name` always be set if using `mode` or `cache`? Does sqlite support `sqlite:file?...`? I'd think so, so we should probably throw a warning here if `name` is unset, or maybe default to `phinx`, e.g. `$options['name'] ?? 'phinx'`?

Reply:
it is supported with an empty filename:

https://github.com/php/php-src/pull/7662/files#diff-0223830cccc48601cf8cb19d5bd6dc243d44b24f2c0aca2776c7f15d0ced8878R13

what would a user expect if it's not set via config - i guess leaving it empty would be right?

Reply:
Makes sense to me, so `$options['name'] ?? ''` should be fine?

Reply:
done